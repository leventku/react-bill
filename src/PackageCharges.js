import React from 'react';
import BillGroup from './BillGroup'

function PackageCharges(props) {
  const renderItem = (item) => (
    <li className="list-group-item justify-content-between" key={item.name}>
      <span className="badge badge-primary package-type">{item.type}</span>
      <span className="name">{item.name}</span>
      <span className="cost">Â£{item.cost.toFixed(2)}</span>
    </li>
  )

  return (
    <BillGroup 
      title="Your Package"
      items={props.data.subscriptions}
      itemRenderer={renderItem}
      total={props.data.total}
    />
  )
}

export default PackageCharges;