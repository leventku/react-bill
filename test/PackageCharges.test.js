import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import PackageCharges from '../src/PackageCharges';

function setupComponent() {
  const props = {
    data: {
      subscriptions: [
        { type: 'tv', name: 'Variety with Movies HD', cost: 50.00 },
      ],
    },
  };

  const wrapper = shallow(<PackageCharges {...props} />);

  return {
    props,
    wrapper,
  };
}

describe('PackageCharges component', () => {
  it('should render', () => {
    const { wrapper } = setupComponent();
    expect(wrapper.exists()).toBe(true);
  });

  it('should render its items', () => {
    const { wrapper } = setupComponent();
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
