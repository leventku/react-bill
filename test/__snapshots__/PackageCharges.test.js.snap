// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PackageCharges component should render its items 1`] = `
<BillGroup
  itemRenderer={[Function]}
  items={
    Array [
      Object {
        "cost": 50,
        "name": "Variety with Movies HD",
        "type": "tv",
      },
    ]
  }
  title="Your Package"
/>
`;
